fish-rust/src/signal.rs

```diff
@@ -4,13 +4,14 @@ use crate::common::{exit_without_destructors, restore_term_foreground_process_gr
 use crate::event::{enqueue_signal, is_signal_observed};
 use crate::nix::getpid;
 use crate::reader::{reader_handle_sigint, reader_sighup};
+use crate::set_errno;
 use crate::termsize::TermsizeContainer;
 use crate::topic_monitor::{generation_t, topic_monitor_principal, topic_t, GenerationsList};
 use crate::wchar::prelude::*;
 use crate::wchar_ffi::{AsWstr, WCharToFFI};
 use crate::wutil::{fish_wcstoi, perror};
 use cxx::{CxxWString, UniquePtr};
-use errno::{errno, set_errno};
+use nix::errno::Errno;
```

Comment:
> If using more of nix is welcome, then let me know if enabling more of its feature flags like fs/signals/user is also fine,

Sure, we should try to increase the level of abstraction.
Do nix APIs depend on `Errno`? That would be an argument for it.